src/code/z_schedule.c

```diff
@@ -0,0 +1,282 @@
+#include "global.h"
+
+#define SCHEDULE_CALC_TIME(hour, minute, dest, temp) \
+    (temp) = (hour)*60.0f;                           \
+    (temp) += (minute);                              \
+    (dest) = (temp) * (0x10000 / 60 / 24.0f);        \
+    (dest) = SCHEDULE_CONVERT_TIME(dest);
+
+s32 Schedule_CheckFlagS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdCheckFlagS* cmd = (ScheduleCmdCheckFlagS*)*script;
+    u16 flag = (cmd->flagByte << 8) | cmd->flagMask;
+
+    if (gSaveContext.weekEventReg[flag >> 8] & (flag & 0xFF)) {
+        *script = *script + cmd->offset;
+    }
+
+    return false;
+}
+
+s32 Schedule_CheckFlagL(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdCheckFlagL* cmd = (ScheduleCmdCheckFlagL*)*script;
+    u16 flag = (cmd->flagByte << 8) | cmd->flagMask;
+
+    if (gSaveContext.weekEventReg[flag >> 8] & (flag & 0xFF)) {
+        *script = *script + (s16)((cmd->offsetH << 8) | cmd->offsetL);
+    }
+
+    return false;
+}
+
+s32 Schedule_CheckTimeRangeS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    s32 inRange = 0;
+    ScheduleCmdCheckTimeRangeS* cmd = (ScheduleCmdCheckTimeRangeS*)*script;
+    f32 f;
+    u16 start;
+    u16 end;
+    u16 now;
+
+    SCHEDULE_CALC_TIME(cmd->startHr, cmd->startMin, start, f);
+
+    SCHEDULE_CALC_TIME(cmd->endHr, cmd->endMin, end, f);
+    end -= 1;
+
+    now = SCHEDULE_TIME_NOW;
+
+    if ((start <= now) && (now <= end)) {
+        inRange = 1;
+    }
+
+    if (inRange == 1) {
+        *script = *script + cmd->offset;
+    }
+
+    return false;
+}
+
+s32 Schedule_CheckTimeRangeL(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    s32 inRange = false;
+    ScheduleCmdCheckTimeRangeL* cmd = (ScheduleCmdCheckTimeRangeL*)*script;
+    f32 f;
+    u16 start;
+    u16 end;
+    u16 now;
+
+    SCHEDULE_CALC_TIME(cmd->startHr, cmd->startMin, start, f);
+
+    SCHEDULE_CALC_TIME(cmd->endHr, cmd->endMin, end, f);
+    end -= 1;
+
+    now = SCHEDULE_TIME_NOW;
+
+    if ((start <= now) && (now <= end)) {
+        inRange = true;
+    }
+
+    if (inRange == true) { // direct comparison required to match
+        *script = *script + (s16)((cmd->offsetH << 8) | cmd->offsetL);
+    }
+
+    return false;
+}
+
+s32 Schedule_ReturnValueL(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdReturnValueL* cmd = (ScheduleCmdReturnValueL*)*script;
+
+    result->result = (cmd->retH << 8) | cmd->retL; //! @bug result is a u8, value is truncated
+    result->hasResult = true;
```

Comment:
```suggestion

    //! @bug result is a u8, value is truncated
    result->result = (cmd->retH << 8) | cmd->retL;
    result->hasResult = true;
```
I think it would make more sense for this comment to go above? I think that's more common